### DISCUSSION TITLE
Cleaning up the directory structure?

### DISCUSSION THREAD

**Discussion by ascpixi** [CosmosOS/Cosmos#2213] [Ideas]
hi guys!
i feel like many directories in this repository can be moved into new repositories to clean-up the structure, some examples:

Root folder:
- `../Demos` -> `CosmosOS/cosmos-examples`
- `../Docs` -> `CosmosOS/cosmos-documentation`
- `../Users` -> either move the contents to separate repositories of owners (i don't think git repositories should contain personal folders...) or remove, as everything there is about 4-5 years old

Source folder:
- `../source/Archive` - remove, i think unused deprecated code should be contained in either its own repo or not included with the stable releases
- `../source/Cosmos.*.*` -> `../source/Cosmos/*/*` - i feel like this structure would work much better; additionally, renaming the folder names to the namespace names would clear up a lot of confusion for newcomers to the project (like me, hehe ^^'>)

i'm not really familiar with the COSMOS file structure, so i'm posting this to Discussions as a casual conversation, but if you guys think this is a good idea, i think we can put this as an issue!

**Comment by quajak** [comment]
In general, I agree with your sentiment.  The users and archive directory can be deleted since its no longer used and increase the size of the project. The change of the source structure also makes sense, although I don't see the advantage of having the extra directories. For sure, the names should match the namespace names. I don't really see the advantage of moving the demos and docs into separate repositories, since the setup is similar to what I have seen in other projects and the clarity gained is less than the extra work going into organization. 

If someone wants to actually put in the work and clean up the repository, that would be greatly appreciated.

**Reply by ascpixi** [reply]
i proposed moving the docs and demos into a seperate repo because it can be used to easily integrate GitHub Pages or Netlify (Netlify is free of charge!), and that's what Microsoft does for things like [Azure](https://github.com/Azure/portaldocs), their [IDEs](https://github.com/microsoft/vscode-docs), [driver samples](https://github.com/microsoft/Windows-driver-samples), and even .NET itself ([samples](https://github.com/dotnet/samples), [docs](https://github.com/dotnet/docs))

i wanted to move the code files in `source` into extra directories because I thought they were all under the same `.csproj`, but they're not, so sorry about that, I feel like keeping it the way in its current state is better :P

i would be happy to make a PR making these file structure changes, but it would require a bit of prodding and trial and error in order to figure out which files are still referenced by components that are being actively used, haha

**Reply by quajak** [reply]
Since we currently already have a github pages action setup for this repo and we need to combine the static docs with the generated documentation, I would prefer keeping them in this repo since I am not sure if the effort is worth it in that case. If I am wrong on this, I am happy to change my mind.

It would be awesome if you went ahead and made the PR. I am happy to answer any further questions you have/if you run into issues with the build processes. We might even be able to reduce the number of .sln files lying around.

**Reply by MishaTy** [reply]
I will do this as the pr is not yet open
